temporalnexus/operation.go

```diff
@@ -22,11 +22,32 @@ import (
 
 	"github.com/nexus-rpc/sdk-go/nexus"
 	"go.temporal.io/api/common/v1"
+
 	"go.temporal.io/sdk/client"
 	"go.temporal.io/sdk/internal"
+	"go.temporal.io/sdk/internal/common/metrics"
+	"go.temporal.io/sdk/log"
 	"go.temporal.io/sdk/workflow"
 )
 
+// GetMetricsHandler returns a metrics handler to be used in a Nexus operation's context.
+func GetMetricsHandler(ctx context.Context) metrics.Handler {
+	nctx, ok := internal.NexusOperationContextFromGoContext(ctx)
+	if !ok {
+		panic("temporalnexus GetMetricsHandler: Not a valid Nexus context")
```

Comment:
I wonder if it would be easier if `NexusOperationContextFromGoContext` did the check for a valid nexus context? That is how the corresponding activity call works. I don't have a very strong opinion here.

Reply:
I also don't have a strong opinion but I went this route because the other usages of `NexusOperationContextFromGoContext` return an error rather than panic and I didn't want `GetMetricsHandler` or `GetLogger` to return an error to keep them parallel to the activity and workflow versions.

Reply:
Yeah at some point will probably need to add a `IsNexusContext` or something like we have for Activities